src/pages/Countries/components/CountriesDisplay/hooks/useFetchCountries.ts

```diff
@@ -0,0 +1,30 @@
+import { Country } from '@types/types';
+import { useFetch } from '../../../hooks/useFetch.ts';
+import { useDebounce } from '../../../hooks/useDebounce.ts';
+const apiConfig = {
+  BASE_URL: 'https://restcountries.com/v3.1',
+  ENDPOINTS: {
+    ALL: 'all',
+    BY_NAME: 'name',
+  },
+  FIELDS: 'name,population,region,capital,flags,cca3',
+} as const;
+
+export function useFetchCountries(searchValue: string) {
+  const debouncedSearch = useDebounce(searchValue);
+
+  const getApiUrl = (search: unknown) => {
```

Comment:
to warto wyrzucić poza hooka (nad niego) albo dorzucić useCallback - chodzi o to, że każde wywołanie tego hooka będzie tworzyło nową funkcję getApiUrl (funkcja będzie robić to samo ale referencja będzie inna), będzie to ciut nadmiarowa operacja